src/tqec/compile/detectors/compute.py

```diff
@@ -494,13 +496,69 @@ def _compute_superimposed_template_instantiations(
     return ret
 
 
+def _extract_subtemplates_from_s3d(
+    s3d: npt.NDArray[numpy.int_],
+) -> list[npt.NDArray[numpy.int_]]:
+    """Extract 2D spatial subtemplates from a 3D array.
+
+    This function takes a 3D array where the first two dimensions
+    represent spatial coordinates and the third dimension represents time steps.
+    It extracts a list of 2D spatial arrays, one for each time step.
+
+    Args:
+        s3d: 3D numpy array representation of sub-templates.
+
+    Returns:
+        List of 2D numpy arrays, one for each time step.
+    """
+    return [s3d[:, :, i] for i in range(s3d.shape[2])]
+
+
+def _compute_detector_for_subtemplate(
+    args: tuple[
+        tuple[int, ...],
+        npt.NDArray[numpy.int_],
+        Sequence[Plaquettes],
+        Shift2D,
+        DetectorDatabase | None,
+        bool,
+    ],
+) -> tuple[tuple[int, ...], frozenset[Detector]]:
+    """Helper function for parallel processing of detector computation.
+
+    Args:
+        args: A tuple containing:
+            - indices: Tuple of indices identifying the subtemplate
+            - s3d: 3D numpy array representing the subtemplate
+            - plaquettes: Sequence of plaquettes for each time slice
+            - increments: Spatial increments between plaquette origins
+            - database: Detector database (optional)
+            - only_use_database: Whether to only use the database
+
+    Returns:
+        A tuple containing the indices and the computed detectors
+    """
+    indices, s3d, plaquettes, increments, database, only_use_database = args
+    return (
+        indices,
+        compute_detectors_at_end_of_situation(
+            _extract_subtemplates_from_s3d(s3d),
+            plaquettes,
+            increments,
+            database,
+            only_use_database,
+        ),
+    )
+
+
 def compute_detectors_for_fixed_radius(
     templates: Sequence[Template],
     k: int,
     plaquettes: Sequence[Plaquettes],
     fixed_subtemplate_radius: int = 2,
     database: DetectorDatabase | None = None,
     only_use_database: bool = False,
+    parallel: bool = False,
```

Comment:
I would replace that argument with something like `parallel_processes_count: int = 1` with the following behaviour:

- If `parallel_processes_count` is 1, that is a sequential run.
- If `parallel_processes_count > 1`, that is a parallel run with the provided number of processes.
- If `parallel_processes_count == -1`, that means "use all the possible processes, so `cpu_count()`.
- If anything else, that is an invalid value.